crates/adapters/src/integrated/delta_table/test.rs

```diff
@@ -124,16 +128,35 @@ async fn wait_for_count_records(
             .await
             .unwrap();
 
-        let count = datafusion
+        let data = datafusion
             .read_table(table.clone())
             .unwrap()
-            .count()
+            .collect()
             .await
             .unwrap();
 
-        info!("expected output table size: {expected_count}, current size: {count}");
+        let mut result = Vec::new();
+
+        for batch in data.iter() {
+            let deserializer = serde_arrow::Deserializer::from_record_batch(batch).unwrap();
+            let mut records =
+                Vec::<T>::deserialize_with_context(deserializer, &delta_input_serde_config())
+                    .unwrap();
 
-        if count == expected_count {
+            result.append(&mut records);
+        }
+
+        info!(
+            "expected output table size: {}, current size: {}",
+            expected_output.len(),
+            result.len()
+        );
+
+        if result.len() == expected_output.len() {
+            result.sort();
+            let mut expected_output = expected_output.to_vec();
+            expected_output.sort();
+            assert_eq!(result, expected_output);
```

Comment:
could be handy to show a diff if they differ
especially for large tables with small diffs

Reply:
Apparently there's a crate for this `pretty_assertions`. Should give it a try.